### PR TITLE
update moby to 19.03.14

### PR DESCRIPTION
**Pull request by alexeldeib**

### CODE REVIEW

**Comment by bowang-666** [comment]
Is there a moby 19.03.14 already?

**Comment by alexeldeib** [comment]
I'm not sure why it's not tagged on the release page, but the artifacts definitely seem to exist: 

```bash
root@aks-nodepool1-42500205-vmss000000:/# apt show moby-engine
Package: moby-engine
Version: **19.03.14+azure-1**
Priority: optional
Section: admin
Maintainer: Microsoft <support@microsoft.com>
Installed-Size: 109 MB
Depends: moby-containerd (>= 1.3.9), moby-runc (>= 1.0.0~rc10), libc6 (>= 2.8), libdevmapper1.02.1 (>= 2:1.02.97), libseccomp2 (>= 2.1.0)
Recommends: apparmor, ca-certificates, iptables, kmod, moby-cli, pigz, xz-utils
Suggests: aufs-tools, cgroupfs-mount | cgroup-lite, git
Conflicts: docker, docker-ce, docker-ee, docker-engine, docker-engine-cs, docker.io, lxc-docker, lxc-docker-virtual-package
Replaces: docker, docker-ce, docker-ee, docker-engine, docker-engine-cs, docker.io, lxc-docker, lxc-docker-virtual-package
Homepage: https://github.com/moby/moby
Download-Size: 22.7 MB
APT-Sources: https://packages.microsoft.com/ubuntu/18.04/prod bionic/main amd64 Packages
Description: Docker container platform (engine package)
  Moby is an open-source project created by Docker to enable and accelerate software containerization.
```

**Comment by alexeldeib** [comment]
the oss repo has it published: https://github.com/moby/moby/releases/tag/v19.03.14

@cpuguy do you know why Azure/moby doesn't have a v19.03.14 release tag? Seems like the artifacts are on packages.microsoft.com and the same version was tagged a while back on moby/moby, any reason we shouldn't be using those?

File: pkg/agent/testdata/RawUbuntuContainerd/line23.sh [hunk 1]
```diff
@@ -105,7 +105,7 @@ installContainerRuntime() {
 
 installMoby() {
     CURRENT_VERSION=$(dockerd --version | grep "Docker version" | cut -d "," -f 1 | cut -d " " -f 3 | cut -d "+" -f 1)
-    local MOBY_VERSION="19.03.12"
+    local MOBY_VERSION="19.03.14"
```
Comment (eriksywu) [inline pkg/agent/testdata/RawUbuntuContainerd/line23.sh:108]: why is this the only test data change?

**Comment by bowang-666** [comment]
lgtm

File: pkg/agent/testdata/RawUbuntuContainerd/line23.sh (hunk 1 above)
Comment (alexeldeib) [reply pkg/agent/testdata/RawUbuntuContainerd/line23.sh:108]: discussed offline, looks like containerd testdata for this case isn't updating. will leave it as-is and let Erik take a look later. naive attempt to fix it didn't seem to produce correct testdata

File: vhdbuilder/packer/install-dependencies.sh [hunk 2]
```diff
@@ -70,7 +70,7 @@ if [[ ${CONTAINER_RUNTIME:-""} == "containerd" ]]; then
   downloadTeleportdPlugin ${TELEPORTD_PLUGIN_DOWNLOAD_URL} "0.5.0"
 else
   CONTAINER_RUNTIME="docker"
-  MOBY_VERSION="19.03.12"
+  MOBY_VERSION="19.03.14"
```
Comment (alexeldeib) [inline vhdbuilder/packer/install-dependencies.sh:73]: required so the var is defined below, but we can definitely clean up the variable handling here.

**Pull request merged**